plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java

```diff
@@ -1117,6 +1117,20 @@ public boolean switchPresentation(ResultSetPresentationDescriptor selectedPresen
             setActivePresentation(instance);
             instance.refreshData(true, false, false);
 
+            try {
+                String presentationId = selectedPresentation.getId();
+                Map<String, Object> configurationMap = Map.of("driver", getDataSource().getContainer().getDriver().getPreconfiguredId());
+                if ("spreadsheet".equals(presentationId)) {
+                    DataEditorFeatures.RESULT_SET_REPRESENTATION_GRID.use(configurationMap);
+                } else if ("plaintext".equals(presentationId)) {
+                    DataEditorFeatures.RESULT_SET_REPRESENTATION_TEXT.use(configurationMap);
+                } else if (presentationId != null && presentationId.toLowerCase(Locale.ROOT).contains("chart")) {
```

Comment:
What is this mess? It's not even the correct place to do that.

Each presentation has `IResultSetPresentation#createPresentation`; you can track features there.

Or even better, send the presentation ID as a parameter instead of having a unique feature per presentation. Just a quick reminder that we also have a document, JSON, XML, GIS, and timeseries presentations.